dj_cqrs/management/commands/cqrs_dead_letters.py

```diff
@@ -0,0 +1,127 @@
+
+    def handle(self, *args, **options):
+        self.check_transport()
+        channel, connection = self.init_broker()
+
+        queue_name, dead_letter_queue_name = RabbitMQTransportService.get_consumer_settings()
+        dead_letters_queue = RabbitMQTransportService.declare_queue(
+            channel, dead_letter_queue_name,
+        )
+        dead_letters_count = dead_letters_queue.method.message_count
+        consumer_generator = channel.consume(
+            queue=dead_letter_queue_name,
+            auto_ack=False,
+            exclusive=False,
+        )
+
+        command = options['command']
+        if command == 'retry':
+            self.handle_retry(channel, consumer_generator, dead_letters_count)
+        elif command == 'bulk_dump':
+            self.handle_bulk_dump(consumer_generator, dead_letters_count)
+        elif command == 'purge':
+            self.handle_purge(channel, dead_letter_queue_name, dead_letters_count)
+
+        if not connection.is_closed:
+            connection.close()
+
+    def check_transport(self):
+        if current_transport is not RabbitMQTransport:
+            raise CommandError("Dead letters commands available only for RabbitMQTransport.")
+
+    def init_broker(self):
+        host, port, creds, exchange = RabbitMQTransportService.get_common_settings()
+        connection, channel = RabbitMQTransportService.create_connection(
+            host, port, creds, exchange,
+        )
+
+        queue_name, dead_letter_queue_name = RabbitMQTransportService.get_consumer_settings()
+        RabbitMQTransportService.declare_queue(channel, queue_name)
+        RabbitMQTransportService.declare_queue(channel, dead_letter_queue_name)
+        for cqrs_id, replica_model in ReplicaRegistry.models.items():
+            channel.queue_bind(exchange=exchange, queue=queue_name, routing_key=cqrs_id)
+
+            # Every service must have specific SYNC or requeue routes
+            channel.queue_bind(
+                exchange=exchange,
+                queue=queue_name,
+                routing_key='cqrs.{}.{}'.format(queue_name, cqrs_id),
+            )
+
+        return channel, connection
+
+    def handle_retry(self, channel, consumer_generator, dead_letters_count):
+        self.stdout.write("Total dead letters: {}".format(dead_letters_count))
+        for i in range(1, dead_letters_count + 1):
+            self.stdout.write("Retrying: {}/{}".format(i, dead_letters_count))
+            method_frame, properties, body = next(consumer_generator)
+
+            dct = ujson.loads(body)
+            dct['retries'] = 0
+            if dct.get('expires'):
+                # Message could expire already
+                expires = get_expires_datetime()
+                dct['expires'] = expires.replace(microsecond=0).isoformat()
+            payload = TransportPayload.from_message(dct)
+            payload.is_requeue = True
+
+            RabbitMQTransportService.produce(payload)
```

Comment:
I am totally okay with current implementation, but I have to ask if it makes sense to make better exception handling here (fails on produce, on nack, etc in other places).

Reply:
For dead_letters commands it's redundant.